internal/server/evaluation/evaluator.go

```diff
@@ -275,6 +200,83 @@ func (e *Evaluator) Evaluate(ctx context.Context, r *flipt.EvaluationRequest) (r
 	return resp, nil
 }
 
+// doConstraintsMatch is a utility function that will return if all or any constraints have matched for a segment depending
+// on the match type.
+func doConstraintsMatch(logger *zap.Logger, evalCtx map[string]string, constraints []storage.EvaluationConstraint, segmentMatchType flipt.MatchType) (bool, error) {
```

Comment:
what about `matchConstraints` ?

```suggestion
func matchConstraints(logger *zap.Logger, evalCtx map[string]string, constraints []storage.EvaluationConstraint, segmentMatchType flipt.MatchType) (bool, error) {
```